test/sagas/repository.spec.js

```diff
@@ -1,37 +1,157 @@
 import { expectSaga } from 'redux-saga-test-plan';
+import { runSaga } from 'redux-saga';
+import { select } from 'redux-saga/effects';
+import moment from 'moment';
 
-import repository, { getDatasets, getStudies } from 'sagas/repository';
+import {
+  authGet,
+  authPost,
+  checkToken,
+  createDataset,
+  getDatasets,
+  getDatasetById,
+  getStudies,
+  getStudyById,
+  getTokenExpiration,
+  getToken,
+} from 'sagas/repository';
 import { ActionTypes } from 'constants/index';
 
 jest.mock('axios', () => ({
-  get: () => ({ items: [] }),
+  get: () => ({ data: { job_status: 'succeeded' } }),
+  post: () => ({ data: { id: 'keep me posted', job_status: 'succeeded' } }),
 }));
 
 describe('repository', () => {
-  it('should have the expected watchers', done =>
-    expectSaga(repository)
-      .run({ silenceTimeout: true })
-      .then(() => {
-        done();
-      }));
-
-  it('should have the get datasets saga', () =>
+  it('test token expiration', () => {
+    const tokenExpiration = 1553537329627;
+    const dispatched = [];
+    const test = runSaga(
+      {
+        dispatch: action => dispatched.push(action),
+        getState: () => ({ user: { tokenExpiration } }),
+      },
+      checkToken,
+    );
+    expect(test.result()).toBe(true);
+  });
+
+  it('test token invalid', () => {
+    const tokenExpired = moment() - 360000;
+    const dispatched = [];
+    const test = runSaga(
+      {
+        dispatch: action => dispatched.push(action),
+        getState: () => ({ user: { tokenExpired } }),
+      },
+      checkToken,
+    );
+    expect(test.result()).toBe(false);
+  });
+
+  it('test get with auth header', () => {
+    const token = 'Cherlene';
+    const tokenExpiration = 1553537329627;
+    const dispatched = [];
+    const test = runSaga(
+      {
+        dispatch: action => dispatched.push(action),
+        getState: () => ({ user: { token, tokenExpiration } }),
+      },
+      authGet,
+    );
+    expect(test.result()).toEqual({ data: { job_status: 'succeeded' } });
+  });
+
+  it('test post with auth header', () => {
+    const token = 'Cherlene';
```

Comment:
For both of these it would be nice to test that axios was getting the Authorization header set correctly